assets/js/blocks/cart-checkout/checkout-i2/frontend.tsx

```diff
@@ -36,12 +40,16 @@ const Wrapper = ( {
 	// we need to pluck out receiveCart.
 	// eslint-disable-next-line no-unused-vars
 	const { extensions, receiveCart, ...cart } = useStoreCart();
+	const checkoutSubmitData = useCheckoutSubmit();
+	const checkoutExtensionData = useCheckoutExtensionData();
```

Comment:
Not 100% confident about the naming of these variables, is it descriptive enough about what it contains?

Reply:
The reason I am choosing to pass the entire object returned by the hook is because how can we know what the consuming components will need?

Reply:
We should skip exposing `useCheckoutSubmit`. It exposes irrelevant data and it's an internal hook that we use to aid showing the submit button, we might give up this hook for other ways to handle the submit button later and having it on public API causes a support liability. 

I see in https://github.com/opr/newsletter-test/blob/main/assets/js/checkout-newsletter-subscription-block/block.js#L10 that you use it for `isDisabled` which IMO something we can skip anyway because clicking the checkbox again wouldn’t really effect a checkout at the submit process.

If it proves that we need `isDisabled`, we can look into exposing Checkout lifecycle related in a dedicated hook, probably a subset of what's exposed from `useCheckoutContext` but without `dispatchActions`.